### PR TITLE
Enable support for Pycharm using the built-in web server

### PR DESCRIPTION
**Pull request by rnw159**
Pycharm has a built-in webserver that makes it easy to open files. I have been using this locally for a while and decided to make a pr to share my work.

Cheers!

There might be better ways to do this, but this works really well for me.

### CODE REVIEW

File: src/popup.html [hunk 1]
```diff
@@ -17,6 +17,7 @@ <h3>Open GitHub links in your IDE</h3>
       <option value="phpstorm">PhpStorm</option>
       <option value="intellij-idea">IntelliJ IDEA</option>
       <option value="webstorm">WebStorm</option>
+      <option value="pycharm">Pycharm (Built-in web server)</option>
```
Comment (lmichelin) [inline src/popup.html:20]:
```suggestion
      <option value="pycharm">PyCharm (built-in web server)</option>
```

File: src/inject.ts [hunk 2]
```diff
@@ -11,43 +11,72 @@ const run = async () => {
...
+      }
+    },
+    pycharm: {
+      name: "Pycharm (Built-in web server)",
+      icon: "icons/intellij-idea32.png",
+      openIDE: (repo: string, file: string, line?: string) => {
+        const url = `http://localhost:63342/api/file?file=${OPTIONS.localPathForRepositories}/${repo}/${file}&line=${line ? `${line}` : "1"}`
+        const Http = new XMLHttpRequest();
+        Http.open("GET", url)
+        Http.send();
```
Comment (lmichelin) [inline src/inject.ts:77]: fetch(url)

**Review by lmichelin — CHANGES_REQUESTED** [review]

File: README.md [hunk 3]
```diff
@@ -19,6 +19,7 @@ This browser extension allows you to open files in your IDE directly from GitHub
 - VS Code Insiders
 - PhpStorm
 - IntelliJ IDEA
+- Pycharm (Built-in web server)
```
Comment (lmichelin) [inline README.md:22]:
```suggestion
- PyCharm (built-in web server)
```

File: src/inject.ts [hunk 4]
```diff
@@ -11,43 +11,72 @@ const run = async () => {
...
+        const url = `webstorm://open?file=${OPTIONS.localPathForRepositories}/${repo}/${file}${line ? `:${line}` : ""}`
+        location.href = url
+        return url
+      }
+    },
+    pycharm: {
+      name: "Pycharm (Built-in web server)",
+      icon: "icons/intellij-idea32.png",
+      openIDE: (repo: string, file: string, line?: string) => {
+        const url = `http://localhost:63342/api/file?file=${OPTIONS.localPathForRepositories}/${repo}/${file}&line=${line ? `${line}` : "1"}`
```
Comment (lmichelin) [inline src/inject.ts:74]:
```suggestion
        const url = `http://localhost:63342/api/file?file=${OPTIONS.localPathForRepositories}/${repo}/${file}&line=${line ?? "1"}`
```

File: README.md [hunk 5]
```diff
@@ -28,6 +29,8 @@ PhpStorm & IntelliJ IDEA support is native on MacOS, but on Linux and Windows yo
 
 I have not tested WebStorm support yet, feedbacks are welcome!
 
+If you wish to use Pycharm (Built-in web server) then you must enable the intellij built-in web server (see [here](https://www.jetbrains.com/help/idea/php-built-in-web-server.html#configuring-built-in-web-server)) on port 63342
```
Comment (lmichelin) [inline README.md:32]:
```suggestion
If you wish to use PyCharm (Built-in web server) then you must enable the intellij built-in web server (see [here](https://www.jetbrains.com/help/idea/php-built-in-web-server.html#configuring-built-in-web-server)) on port 63342.
```

Btw I think that this line can go after line 25 ;)

**Comment by lmichelin** [comment]
Thanks @rnw159 for your contribution :wink: 

I made some updates to your PR in commit 868fc5d:
- replace "PyCharm" with "JetBrains" as the build-in web server is available in many JetBrains editors
- add a JetBrains icon
- use `fetch` instead of `XMLHttpRequest`
- show an alert if the API call fails
- refactor some code

I just published v1.1.7 to Chrome Web Store and Firefox Addons.

**Comment by rnw159** [comment]
Hey thanks for merging! I love this extension 💚

**Pull request merged**